src/java/picard/illumina/ClusterDataToSamConverter.java

```
@@ -101,10 +107,39 @@ public ClusterDataToSamConverter(final String runBarcode,
     }
 
     /**
+     *  Sets the SAM tag to use to store the molecular index bases.  If multiple molecular indexes exist, it will concatenate them
+     * and store them in this tag.
+     */
+    public ClusterDataToSamConverter withMolecularIndexTag(final String molecularIndexTag) {
+        this.molecularIndexTag = molecularIndexTag;
+        return this;
+    }
+
+    /**
+     * Sets the SAM tag to use to store the molecular index base qualities.  If multiple molecular indexes exist, it will concatenate them
+     * and store them in this tag.
+     */
+    public ClusterDataToSamConverter withMolecularIndexQualityTag(final String molecularIndexQualityTag) {
+        this.molecularIndexQualityTag = molecularIndexQualityTag;
+        return this;
+    }
+
+    /**
+     * Sets the SAM tags to use to store the bases each molecular index.  This will only be used if there are more than one molecular
+     * index. If fewer tags are given than molecular indexes found, then the remaining molecular indexes will be concatenated and stored
+     * in the last tag.  If more tags are provided than molecular indexes found, the additional tags will not be used.
+     */
+    public ClusterDataToSamConverter withTagPerMolecularIndex(final List<String> tagPerMolecularIndex) {
+        if (tagPerMolecularIndex != null) this.tagPerMolecularIndex = tagPerMolecularIndex;
```

Should this throw a IllegalArgumentException rather than silently ignore a null being passed in?